cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -0,0 +1,363 @@
+package appdetect
+
+import (
+	"fmt"
+	"log"
+	"maps"
+	"slices"
+	"strings"
+)
+
+type SpringBootProject struct {
+	springBootVersion     string
+	applicationProperties map[string]string
+	parentProject         *mavenProject
+	mavenProject          *mavenProject
+}
+
+type DatabaseDependencyRule struct {
+	databaseDep       DatabaseDep
+	mavenDependencies []MavenDependency
+}
+
+type MavenDependency struct {
+	groupId    string
+	artifactId string
+}
+
+var databaseDependencyRules = []DatabaseDependencyRule{
+	{
+		databaseDep: DbPostgres,
+		mavenDependencies: []MavenDependency{
+			{
+				groupId:    "org.postgresql",
+				artifactId: "postgresql",
+			},
+		},
+	},
+	{
+		databaseDep: DbMySql,
+		mavenDependencies: []MavenDependency{
+			{
```

Comment:
nit: Just provide another angle to see this, jdbc driver is a dependency common to all Java projects, not only Spring projects. So we might alter this later in the future, if we want to remodel it for Java, Spring Framework, and Quarkus Framework.